nfs/k8s/k8s.go

```diff
@@ -132,16 +140,10 @@ func (kc *K8sClient) GetNodeByCSINodeId(ctx context.Context, driverKey string, c
 	if err != nil {
 		return nil, fmt.Errorf("failed to list nodes: %w", err)
 	}
+
 	for _, node := range nodeList.Items {
-		if annotation, exists := node.Annotations["csi.volume.kubernetes.io/nodeid"]; exists {
```

Comment:
I don't see how this code will accomplish the same outcome. The "csi.volume.kubernetes.io/nodeid" annotation will contain multiple driverKeys -> systemIds. There isn't a single annotation based on the driverKey.

Reply:
I think Luke took this from the changes I had for k8s.go. It appears that node.Annotations is a map already with the driverKey being the key. In this code, we were getting a string and unmarshalling it and It would fail all the time because it was not in JSON format.  I had to make this change for unit test, but I will check again to see if anything is missed.